geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/file/FileBlobStore.java

```diff
@@ -696,15 +789,108 @@ private Properties getLayerMetadata(final String layerName) {
                         log.warn(e.getMessage(), e);
                     }
                 }
+            } else {
+                log.error(
+                        "Uncompressed metadata file does not exists: " + metadataFile.getPath());
             }
         }
         return properties;
     }
 
+    private Properties getLayerMetadata(final File metadataFile) {
+        Properties properties = new Properties();
+        long lastModified = metadataFile.lastModified();
+        final String lockObj = metadataFile.getAbsolutePath().intern();
+        synchronized (lockObj) {
+            if (metadataFile.exists()) {
+                FileInputStream in;
+                GZIPInputStream gzipIn;
+                try {
+                    in = new FileInputStream(metadataFile);
+                    gzipIn = new GZIPInputStream(in);
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+                try {
+                	int tryagain = 0;
+                    while (tryagain < FileBlobStore.METADATA_MAX_RW_ATTEMPTS) {
+                        // Read current metadatafile date modification (currentModTime)
+                        long currentModDate = metadataFile.lastModified();
+	                	if (lastModified == currentModDate) {
```

Comment:
It's safer to first read and then check if the last modified date changed, ensures that nobody modified the file during the read.